testing/profiler-tests/src/test/java/com/splunk/opentelemetry/ProfilerSmokeTest.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.splunk.opentelemetry;
+
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Spliterator;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.StreamSupport;
+import org.junit.ClassRule;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.containers.BindMode;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.images.builder.ImageFromDockerfile;
+import org.testcontainers.utility.MountableFile;
+
+public class ProfilerSmokeTest {
+
+  @ClassRule public static GenericContainer<?> petclinic;
+
+  static {
+    MountableFile agentJar = MountableFile.forHostPath(findAgentJar());
+    petclinic =
+        new GenericContainer<>(new ImageFromDockerfile().withDockerfile(Path.of("./Dockerfile")))
+            .withExposedPorts(9966)
+            .withCopyFileToContainer(agentJar, "/app/javaagent.jar")
+            .withCreateContainerCmdModifier(cmd -> cmd.withEntrypoint("java"))
+            .withCommand(
+                "-javaagent:/app/javaagent.jar",
+                "-Dotel.javaagent.debug=true",
+                "-Dsplunk.profiler.enabled=true",
+                "-Dsplunk.profiler.directory=/app/jfr",
+                "-jar",
+                "/app/spring-petclinic-rest.jar")
+            .withFileSystemBind("build/test/output", "/app/jfr", BindMode.READ_WRITE)
```

Comment:
Hmm, if you put this code in a `@BeforeAll` you'd be able to use JUnit's `@TempDir` instead